packages/core/src/common/disposable.ts

```diff
@@ -47,6 +47,29 @@ Object.defineProperty(Disposable, 'NULL', {
     }
 });
 
+/**
+ * Utility for tracking a collection of Disposable objects, specifically for
+ * the sake of finding out when all items in the collection have been disposed.
+ *
+ * ```
+ * const dc = new DisposableCollection(myDisposables);
+ * dc.onDispose(() => {
+ *    console.log('All elements in the collection have been disposed');
```

Comment:
I'd say it is a bit risky to claim this as the main goal of this class since the `onDispose` element is actually only called once, even if you keep re-using the collection.

Reply:
See my comment above. That it lets you continue using the collection after reaching length zero is an oddity for sure. But I stand by the claim. See my argument above and let me know if you're still not convinced.

Reply:
See answer above ;-) And yes, I agree it is an oddity and I'm not sure why it was done this way but to have it documented is definitely the right move, especially since it may not be intuitive.